### ISSUE TITLE
small issue about quantization in carlini.py

### ISSUE THREAD

**Issue by ZhengyuZhao** [jeromerony/fast_adversarial#6]
Hi,

I run the C&W attack with carlini.py and notice that in some cases, the adversarial output will be out of the range of [0,1] when setting the argument "quantize=True".

It turned out that a clamping operation is needed in the _quantize function since this function is only called after all the iterations.  Am I right?

**Comment by jeromerony** [comment]
Hi,

You are perfectly right. 

For context, I implemented this function following a discussion with Carlini who explained to me how he did it, but did not have the original code anymore. However, we ended up not using it for the article so I stopped modifying it and I think it should be removed from the repo for a number of reasons (the function is not correct, it does not take the number of levels for quantization, etc.).

If you really want to use it though, you'd need to correct it so that the [step](https://github.com/jeromerony/fast_adversarial/blob/1316409f34531422a37b98f4f570103c1a62dd19/fast_adv/attacks/carlini.py#L243) does not go out of [0, 1]. The non-trivial part is that you need to check that `0 <= delta - multiplier * direction / 255 <= 1` and if it's not the case, update `direction` accordingly.

TLDR; don't use it

**Comment by ZhengyuZhao** [comment]
Thanks for your explanation!
